File: Cpp/1103_distribute_candies_to_people.cpp
// https://leetcode.cn/problems/distribute-candies-to-people/

class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        vector<int> res(num_people, 0);
        int idx = 0, candy = 1;
        while (candies > 0) {
            candy = min(candy, candies);
            res[idx] += candy;
            candies -= candy;

            candy += 1;
            idx = (idx + 1) % num_people;
        }
        return res;
    }
};

File: Python3/1103_distribute_candies_to_people.py
# https://leetcode.cn/problems/distribute-candies-to-people/

class Solution:
    def distributeCandies(self, candies, num_people):
        """
        Inputs:
            candies, num_people: int
        Outputs:
            res: list[int]
        """
        res = [0 for _ in range(num_people)]
        candy, idx = 1, 0
        while candies > 0:
            candy = min(candy, candies)
            res[idx] += candy
            candies -= candy

            candy += 1
            idx = (idx + 1) % num_people
        return res